freqtrade/strategy/backtest_lookahead_bias_checker.py

```diff
@@ -0,0 +1,226 @@
+        return result
+
+    @staticmethod
+    def report_signal(result, column_name, checked_timestamp):
+        df = result['results']
+        row_count = df[column_name].shape[0]
+
+        if row_count == 0:
+            return False
+        else:
+
+            df_cut = df[(df[column_name] == checked_timestamp)]
+            if df_cut[column_name].shape[0] == 0:
+                # print("did NOT find the same signal in column " + column_name +
+                #       " at timestamp " + str(checked_timestamp))
+                return False
+            else:
+                return True
+        return False
+
+    # analyzes two data frames with processed indicators and shows differences between them.
+    def analyze_indicators(self, full_vars, cut_vars, current_pair):
+        # extract dataframes
+        cut_df = cut_vars.indicators[current_pair]
+        full_df = full_vars.indicators[current_pair]
+
+        # cut longer dataframe to length of the shorter
+        full_df_cut = full_df[
+            (full_df.date == cut_vars.compared_dt)
+        ].reset_index(drop=True)
+        cut_df_cut = cut_df[
+            (cut_df.date == cut_vars.compared_dt)
+        ].reset_index(drop=True)
+
+        # compare dataframes
+        if full_df_cut.shape[0] != 0:
+            if cut_df_cut.shape[0] != 0:
+                compare_df = full_df_cut.compare(cut_df_cut)
+
+                # skippedColumns = ["date", "open", "high", "low", "close", "volume"]
+                for col_name, values in compare_df.items():
+                    col_idx = compare_df.columns.get_loc(col_name)
+                    compare_df_row = compare_df.iloc[0]
+                    # compare_df now comprises tuples with [1] having either 'self' or 'other'
+                    if 'other' in col_name[1]:
+                        continue
+                    self_value = compare_df_row[col_idx]
+                    other_value = compare_df_row[col_idx + 1]
+
+                    # output differences
+                    if self_value != other_value:
+
+                        if not self.current_analysis.false_indicators.__contains__(col_name[0]):
+                            self.current_analysis.false_indicators.append(col_name[0])
+                            print(f"=> found look ahead bias in indicator {col_name[0]}. " +
+                                  f"{str(self_value)} != {str(other_value)}")
+
+    def prepare_data(self, varHolder, pairs_to_load):
+        prepare_data_config = copy.deepcopy(self.local_config)
+        prepare_data_config['timerange'] = (str(self.dt_to_timestamp(varHolder.from_dt)) + "-" +
+                                            str(self.dt_to_timestamp(varHolder.to_dt)))
+        prepare_data_config['pairs'] = pairs_to_load
+        self.backtesting = Backtesting(prepare_data_config)
+        self.backtesting._set_strategy(self.backtesting.strategylist[0])
+        varHolder.data, varHolder.timerange = self.backtesting.load_bt_data()
```

Comment:
> another in-between-state. Will add more functionality now.
> Currently when testing with futures-data it fails and don't know why (yet)

from afar, i'd say because you're missing the call to `.load_bt_data_detail()`...

it's a guess - but since you're not using `.start()` - a pretty good one :laughing: